### ISSUE TITLE
Option to disable install rpath

### ISSUE THREAD

**Issue by traversaro** [robotology/how-to-export-cpp-library#7]
In the template, we recommend to add a new option `ENABLE_RPATH` to control whether or not enable rpath on the installation binaries : https://github.com/robotology/how-to-export-cpp-library/blob/master/CMakeLists.txt#L57 . 

I noticed that now YARP deprecated the option for doing that, and instead is using the CMake existing option `CMAKE_SKIP_INSTALL_RPATH` to control if enabling installation rpath https://github.com/robotology/yarp/blob/eeeb98345d528ef56187b06c3712fdd66fadeb08/doc/installation/install_linux.dox#L165 . Using an already existing CMake variable seems to be a good idea. 
I propose to do the same here, but listing `CMAKE_SKIP_INSTALL_RPATH` explicitly as an option like we do with `BUILD_SHARED_LIBS` and `BUILD_TESTING`.

**Comment by traversaro** [comment]
cc @drdanz @francesco-romano @claudiofantacci

**Comment by traversaro** [comment]
@francesco-romano I would eventually do the same in all the the repo in which we use `AddInstallRPATHSupport`.

**Comment by claudiofantacci** [comment]
I think that it is reasonable. I'm ok for doing this change, using the `CMAKE_SKIP_INSTALL_RPATH` variable.

We should wait CMake version 3.9 ([see out issue on GitLab](https://gitlab.kitware.com/cmake/cmake/issues/16589)) to have a consistent behavior of `CMAKE_SKIP_INSTALL_RPATH` in all platforms.

**Comment by traversaro** [comment]
cc @francesco-romano

**Comment by claudiofantacci** [comment]
I'm finally going through this issue 😄 

There are some things to consider here (maybe in the meantime I got some confused ideas, so bear with me):
 - Should we enforce macOS users to have at least CMake version 3.9 to avoid the problem with RPATH discussed [here](https://gitlab.kitware.com/cmake/cmake/issues/16589)?
 - Should I also add `CMAKE_SKIP_RPATH` in place of `ENABLE_RPATH` _and_ remove the [DEPENDS](https://github.com/robotology/how-to-export-cpp-library/blob/master/CMakeLists.txt#L82) option altogether?

**Comment by claudiofantacci** [comment]
ping @traversaro

**Comment by traversaro** [comment]
My take on this: 

>  Should we enforce macOS users to have at least CMake version 3.9 to avoid the problem with RPATH discussed [here](https://gitlab.kitware.com/cmake/cmake/issues/16589)?

Yes, most of the macOS user have the latest CMake anyway. 

>  Should I also add `CMAKE_SKIP_BUILD_RPATH` in place of `ENABLE_RPATH` _and_ remove the [DEPENDS](https://github.com/robotology/how-to-export-cpp-library/blob/master/CMakeLists.txt#L82) option altogether?

I am  not an expert on the RPATH macro, and I am not super sure about what it is doing w.r.t. to RPATH. However, I agree on relying on existing CMake variables as much as possible, perhaps explicitly exposing them as options as we do with `BUILD_SHARED_LIBS` and `BUILD_TESTING`.

**Comment by claudiofantacci** [comment]
Thanks @traversaro, maybe on this

> > Should I also add CMAKE_SKIP_BUILD_RPATH in place of ENABLE_RPATH and remove the DEPENDS option altogether?
> 
> I am not an expert on the RPATH macro, and I am not super sure about what it is doing w.r.t. to RPATH. However, I agree on relying on existing CMake variables as much as possible, perhaps explicitly exposing them as options as we do with BUILD_SHARED_LIBS and BUILD_TESTING.

@francesco-romano may probide his opninion! 😄

**Comment by francesco-romano** [comment]
I agree on the CMake version.
macOS users usually use Homebrew to manage command line tools and libraries, and this usually implies that they are up-to-date (often they are too much up-to-date!)

Regarding CMake variables:
**TL;DR**: test if the default variables are enough for the behaviour we want to support as I don't remember 

I also agree in avoiding as much as possible the introduction of new variables if CMake provides an alternative. Regarding RPATH, I do not actually remember if we were able to control the behaviour by using CMake variables in an easy way, and I did not work a lot after CMake 3.9 (that changed the behaviour as we asked :D).
The behaviour I usually like to support are only two:
- RPATH enabled: use rpath
- RPATH disabled: use absolute installation path

**Comment by claudiofantacci** [comment]
> I agree on the CMake version.

👍 

> test if the default variables are enough for the behaviour we want to support as I don't remember

I'll make some test, maybe the variable `CMAKE_SKIP_RPATH` can be used in place of `ENABLE_RPATH`.

> The behaviour I usually like to support are only two:
> 
>     RPATH enabled: use rpath
>     RPATH disabled: use absolute installation path

This is ok, it's reasonable, but it means that we will support

 case | `INSTALL_NAME_DIR` | `CMAKE_SKIP_RPATH`/`NOT ENABLE_RPATH` | install_name
----|----------------------|------------------|-------------
1 | `DEFINED` | `TRUE` | /path/foo.dylib
2 | `NOT DEFINED` | `FALSE` | @ rpath/foo.dylib

and that `CMAKE_SKIP_RPATH/NOT ENABLE_RPATH=FALSE` implies `CMAKE_SKIP_BUILD/INSTALL_RPATH=FALSE`.

**Comment by claudiofantacci** [comment]
(updated previous comment)
